### PR TITLE
Bug 1229595 - Find in Page: Animate between results

### PR DESCRIPTION
**Pull request by bkmunar**
1. added a function that adds scroll animation for find in page

### CODE REVIEW

File: Client/Assets/FindInPage.js [hunk 1]
```diff
@@ -162,10 +165,30 @@ function updateHighlightedSpan() {
   var left = rect.left + scrollX - window.innerWidth / 2;
   left = clamp(left, 0, document.body.scrollWidth);
   top = clamp(top, 0, document.body.scrollHeight);
-  window.scrollTo(left, top);
+  scrollToSelection(left, top, 400);
```
Comment (thebnich) [inline Client/Assets/FindInPage.js]: Move the 400 to a constant

File: Client/Assets/FindInPage.js [hunk 2]
```diff
@@ -150,6 +152,7 @@ function getIDForRect(rect) {
 function updateHighlightedSpan() {
   // Reset the color of the previous highlight.
   if (activeHighlightSpan) {
+    clearInterval(scrollInterval);
```
Comment (thebnich) [inline Client/Assets/FindInPage.js]: Why do you have a `clearInterval` in here?

File: Client/Assets/FindInPage.js [hunk 3]
```diff
@@ -162,10 +165,30 @@ function updateHighlightedSpan() {
   var left = rect.left + scrollX - window.innerWidth / 2;
   left = clamp(left, 0, document.body.scrollWidth);
   top = clamp(top, 0, document.body.scrollHeight);
-  window.scrollTo(left, top);
+  scrollToSelection(left, top, 400);
   debug("Scrolled to: " + left + ", " + top);
 }
 
+function scrollToSelection(left, top, duration) {
+  var startTime = 0,
```
Comment (thebnich) [inline Client/Assets/FindInPage.js]: `startTime` -> `time`. It's only the "start" when it's 0 :)

File: Client/Assets/FindInPage.js [hunk 4]
```diff
@@ -162,10 +165,30 @@ function updateHighlightedSpan() {
...
+  startY = scrollY;
+  scrollInterval = setInterval(function() {
+    if (startTime < duration) {
+      var xStep = easeOutCubic(startTime, startX, left - startX, duration),
+      yStep = easeOutCubic(startTime, startY, top - startY, duration);
+      window.scrollTo(xStep, yStep);
+      startTime += SCROLL_INTERVAL_INCREMENT;
+    } else {
+     clearInterval(scrollInterval);
+    }
```
Comment (thebnich) [inline Client/Assets/FindInPage.js]: Nit: replace the `if`/`else` with just an `if`:

```
time += SCROLL_INTERVAL_INCREMENT;
if (time > duration) {
  clearInterval(scrollInterval);
}
```

File: Client/Assets/FindInPage.js (hunk 3 above)
Comment (bkmunar) [inline Client/Assets/FindInPage.js]: omg the smiley why

**Comment by bkmunar** [comment]
done w/ nitz

File: Client/Assets/FindInPage.js [hunk 5]
```diff
@@ -162,10 +165,32 @@ function updateHighlightedSpan() {
...
   top = clamp(top, 0, document.body.scrollHeight);
-  window.scrollTo(left, top);
+  scrollToSelection(left, top, SCROLL_INTERVAL_DURATION);
   debug("Scrolled to: " + left + ", " + top);
 }
 
+function scrollToSelection(left, top, duration) {
+  var time = 0,
+  startX = scrollX,
+  startY = scrollY;
```
Comment (thebnich) [inline Client/Assets/FindInPage.js]: Use `var` on each line; comma-separated declarations are generally considered poor form.

File: Client/Assets/FindInPage.js [hunk 6]
```diff
@@ -162,10 +165,32 @@ function updateHighlightedSpan() {
...
+  clearInterval(scrollInterval);
+  scrollInterval = setInterval(function() {
+    if (time < duration) {
```
Comment (thebnich) [inline Client/Assets/FindInPage.js]: You don't need this -- if you're still in the interval, that means you haven't cleared it yet, so you know time will be <= duration.

File: Client/Assets/FindInPage.js [hunk 7]
```diff
@@ -162,10 +165,32 @@ function updateHighlightedSpan() {
...
+      var xStep = easeOutCubic(time, startX, left - startX, duration),
+          yStep = easeOutCubic(time, startY, top - startY, duration);
+      window.scrollTo(xStep, yStep);
+    }
+    time += SCROLL_INTERVAL_INCREMENT;
+    if (time > duration) {
```
Comment (thebnich) [inline Client/Assets/FindInPage.js]: >=

File: Client/Assets/FindInPage.js [hunk 8]
```diff
@@ -162,10 +165,32 @@ function updateHighlightedSpan() {
...
 
+function scrollToSelection(left, top, duration) {
+  var time = 0,
+  startX = scrollX,
+  startY = scrollY;
+  clearInterval(scrollInterval);
+  scrollInterval = setInterval(function() {
+    if (time < duration) {
+      var xStep = easeOutCubic(time, startX, left - startX, duration),
+          yStep = easeOutCubic(time, startY, top - startY, duration);
```
Comment (thebnich) [inline Client/Assets/FindInPage.js]: Same nit here, use separate `var`s.

**Pull request merged**